subprojects/build-init/src/test/groovy/org/gradle/buildinit/tasks/InitBuildSpec.groovy

```diff
@@ -175,4 +175,30 @@ class InitBuildSpec extends Specification {
         e.message == "Package name is not supported for 'some-type' build type."
     }
 
+    def "should reject invalid package name: #invalidPackageName"() {
+        given:
+        projectLayoutRegistry.get("java-library") >> projectSetupDescriptor
+        projectSetupDescriptor.modularizationOptions >> [ModularizationOption.SINGLE_PROJECT]
+        projectSetupDescriptor.testFrameworks >> [SPOCK]
+        projectSetupDescriptor.dsls >> [GROOVY]
+        projectSetupDescriptor.supportsPackage() >> true
+        init.type = "java-library"
+        init.packageName = invalidPackageName
+
+        when:
+        init.setupProjectLayout()
+
+        then:
+        GradleException e = thrown()
+        e.message == "Package name: '" + invalidPackageName + "' is not valid - it may contain invalid characters or reserved words."
+
+        where:
+        invalidPackageName << [
+            'some.new.thing',
+            'my.package',
+            '2rt9.thing',
```

Comment:
I think `_` is legal, but `-` is not. Can you add another failing example for `com.not-legal` ?